yt_dlp/extractor/rokfin.py

```diff
@@ -0,0 +1,740 @@
+                        self.write_debug(f'{content_type}: could not process content entry: {content}')
+                        return
+
+                    video_data['title'] = str_or_none(traverse_obj(content, ('content_title', 'raw')))
+                    return video_data
+
+                pages_total = int_or_none(traverse_obj(srch_res, ('meta', 'page', 'total_pages')))
+                if pages_total <= 0 or not query:
+                    return []
+                if pages_total is None:
+                    self.report_warning(msg='unknown total # of pages of search results. This may be a bug', only_once=True)
+                elif (pages_total_printed is False) and max_pages_to_download is not None:
+                    self.to_screen(msg=f'Pages to download: {min(pages_total, max_pages_to_download)}')
+                    pages_total_printed = True
+
+                results_total = int_or_none(traverse_obj(srch_res, ('meta', 'page', 'total_results')))
+                if results_total is None:
+                    self.report_warning(msg='unknown total # of search results. This may be a bug', only_once=True)
+                elif results_total_printed is False:
+                    self.to_screen(msg=f'Search results available: {results_total}')
+                    results_total_printed = True
+
+                for content in srch_res.get('results', []):
+                    video_data = get_video_data(content)
+
+                    if not video_data:
+                        continue
+
+                    yield self.url_result(url=video_data.get('url'), video_id=video_data.get('id'), video_title=video_data.get('title'))
+                    yielded_result_counter += 1
+
+                    if yielded_result_counter >= min(n_results, results_total or float('inf')) or (n_results == float('inf') and results_total is None):
+                        # If Rokfin (unexpectedly) does not report the total # of search results,
+                        # and n_results == inf, then the downloading loop has no definitive stopping point
+                        # and could, theoritically, execute indefinitely. To prevent this, we proactively
+                        # quit the loop.
+                        #
+                        # The good news is: this is an unlikely scenario and should not occur routinely.
+                        if n_results == float('inf') and results_total is None:
+                            self.report_warning(msg='please specify a finite number of search results, e.g. 100, and re-run. Stopping the downloading process prematurely to avoid an infinite loop')
+
+                        return
+
+                if page_n >= min(pages_total or float('inf'), max_pages_to_download or float('inf')) or (pages_total is None and max_pages_to_download is None):
+                    return
+
```

Comment:
These kinds of checks are not needed. Override `_search_results` instead of `_get_n_results` and keep yielding results till the API returns a page with no results. The core code will handle the rest